WcaOnRails/app/controllers/delegates_controller.rb

```diff
@@ -7,4 +7,29 @@ class DelegatesController < ApplicationController
   def stats
     @delegates = User.delegates.includes(:actually_delegated_competitions)
   end
+
+  def probations
+    @probation_roles = Role.where(group_id: Group.where(group_type: "delegate_probation"))
+    @probation_users = {}
+    @probation_roles.each { |probation_role|
+      @probation_users[probation_role.user_id] = User.find_by_id(probation_role.user_id)
+    }
+  end
+
+  def start_probation
+    wca_id = params[:wcaId]
+    user = User.find_by_wca_id!(wca_id)
+    role = Role.new(
+      user_id: user.id,
+      group_id: Group.find_by!(name: "Delegate Probation").id,
+      start_date: Date.today,
+    )
+    role.save!
+  end
+
+  def end_probation
+    probation_role_id = params[:probationRoleId]
+    role = Role.find_by_id(probation_role_id)
+    role.update(end_date: Date.today)
```

Comment:
```suggestion
    role.update!(end_date: Date.today)
```

Reply:
The exclamation mark is important because it makes the `update` method throw an error in case the validations fail. The version without `!` simply returns `false`, and since you're not checking the return value it would fail silently

Reply:
Done.